packages/block-library/src/query/edit/query-inspector-controls.js

```diff
@@ -129,8 +129,12 @@ export default function QueryInspectorControls( {
 
 	const [ querySearch, setQuerySearch ] = useState( query.search );
 	const onChangeDebounced = useCallback(
-		debounce( () => setQuery( { search: querySearch } ), 250 ),
-		[ querySearch ]
+		debounce( () => {
```

Comment:
`debounce` makes me always think that there is an underlying problem that can't be easily solved and it makes the issue invisible. Do you know why is it called in `useEffect` in the first place?

Reply:
Does it exist here to prevent too many requests to the server when typing?

Reply:
>Does it exist here to prevent too many requests to the server when typing?

Yes.